### DISCUSSION TITLE
How to add audio and image simultaneously for a fixed time

### DISCUSSION THREAD

**Discussion by ruchirasakalle** [jspsych/jsPsych#3059] [Q&A]
I want to display an image and play a sound at the same time for fixed duration. I checked the psychophysics plugin demos but still a little clueless. As two stimulus are not presented simultaneously.

**Comment by ChristopheBossens** [comment]
Hi ruchirasakalle,

I think the psychophysics plugin is only meant for visual stimulus display. The plugin does use the requestAnimationFrame function for controlling stimulus onset times, which gives you the best possible control for visual stimulus presentation duration. 

My best bet would be to extend the code of that plugin so that it also starts your audio on the first frame of the stimulus, although I can imagine that there could still be some latencies in the onset of the audio so that it will still not be exactly simultaneous.

Alternative, have you tried with the audio response plugins from jsPsych (e.g. https://www.jspsych.org/7.3/plugins/audio-button-response/ )? You can include a visual stimulus in the prompt parameter and also gives you sufficient accuracy?

**Reply by ruchirasakalle** [reply]
I will try to use the visual stimulus as prompt and see how it goes. Thank you so much for your response

**Reply by kassyec** [reply]
I am curious if you ever found a solution to your problem and if you wouldn't mind sharing what you did - I am currently trying to add image + audio on the button response plugin with no success.

**Comment by ruchirasakalle** [comment]
I used the psychophysics plugin and it worked for me. I can share my code
snipped with you. I basically set the same stimulus onset time. For both
audio and image.
